kombu/common.py

```diff
@@ -105,6 +105,12 @@ def declaration_cached(entity, channel):
 
 def maybe_declare(entity, channel=None, retry=False, **retry_policy):
     """Declare entity (cached)."""
+    if retry:
+        return _imaybe_declare(entity, channel, **retry_policy)
+    return _maybe_declare(entity, channel)
+
+
+def _maybe_declare(entity, channel):
     is_bound = entity.is_bound
```

Comment:
@tyarimi I have studied the function and the flows you are describing, and although the changes seem logical, I still don't see the issue that you are describing. Could you please explain it in steps? Thanks.